orc8r/cloud/go/services/bootstrapper/servicers/registration/registration.go

```diff
@@ -114,12 +118,16 @@ func makeErr(errString string) *protos.RegisterResponse {
 }
 
 // createGatewayDevice creates the gateway device model
-func createGatewayDevice(hwID *protos.AccessGatewayID, challengeKey *protos.ChallengeKey) *models.GatewayDevice {
-	challengeKeyBase64 := strfmt.Base64(challengeKey.Key)
+func createGatewayDevice(hwID *protos.AccessGatewayID, challengeKey *protos.ChallengeKey) (*models.GatewayDevice, error) {
+	decodedKey, err := base64.StdEncoding.DecodeString(string(challengeKey.Key))
+	if err != nil {
+		return nil, err
+	}
+	challengeKeyBase64 := strfmt.Base64(decodedKey)
 	return &models.GatewayDevice{
```

Comment:
Nit: I don't like returning literal structs because I prefer always a single-line return for readability. So i.e. create a variable := the struct literal, then return the variable (and nil)